handler/event_handler.go

```diff
@@ -0,0 +1,245 @@
+package handler
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"reflect"
+	"time"
+
+	"github.com/ONSdigital/dp-import-reporter/config"
+	"github.com/ONSdigital/go-ns/log"
+	"github.com/coocood/freecache"
+)
+
+const failed = "failed"
+
+//main function that triggers everything else
+func (e *EventReport) HandleEvent(httpClient *http.Client, c *freecache.Cache, cfg *config.Config) error {
+	log.Info("Starting error handle", log.Data{"INSTANCE_ID": e.InstanceID, "ERROR_MSG": e.EventMsg})
+
+	status, events, err := e.checkInstance(httpClient, cfg)
+	if err != nil {
+		return err
+	}
+	log.Info("Successfully checked instance", log.Data{"INSTANCE_ID": e.InstanceID, "INSTANCE_STATE": status})
+	instanceEvents := &InstanceEvent{
+		Type:          e.EventType,
+		Message:       e.EventMsg,
+		MessageOffset: "0",
+	}
+
+	timeNow := time.Now()
+
+	jsonUpload, err := json.Marshal(Event{
+		Type:          e.EventType,
+		Time:          &timeNow,
+		Message:       e.EventMsg,
+		MessageOffset: "0",
+	})
+
+	key := []byte(e.InstanceID)
+	value := []byte(e.EventMsg)
+	expire := 25
+
+	check := arraySlicing(instanceEvents, events)
+	if check {
+		_, err := c.Get(key)
```

Comment:
As the only thing being returned here is an error, this could be inlined as well (though not exactly the same as comma ok) `if _, err := c.Get(key); err != nil {` - there are a few places this can be done throughout